adam-core/src/main/scala/org/bdgenomics/adam/rdd/features/CoverageRDD.scala

```diff
@@ -0,0 +1,225 @@
+   * @return Gets the SequenceRecord for coverage.
+   */
+  private def getSequenceRecord(elem: Coverage): SequenceRecord = {
+    SequenceRecord(elem.contigName, 1L)
+  }
+
+  /**
+   * Builds a CoverageRDD without SequenceDictionary information by running an
+   * aggregate to rebuild the SequenceDictionary.
+   *
+   * @param rdd The underlying Coverage RDD to build from.
+   * @return Returns a new CoverageRDD.
+   */
+  private[rdd] def apply(rdd: RDD[Coverage]): CoverageRDD = {
+
+    // aggregate to create the sequence dictionary
+    val sd = new SequenceDictionary(rdd.map(getSequenceRecord)
+      .distinct
+      .collect
+      .toVector)
+
+    CoverageRDD(rdd, sd)
+  }
+}
+
+case class CoverageRDD(rdd: RDD[Coverage],
+                       sequences: SequenceDictionary) extends GenomicRDD[Coverage, CoverageRDD] {
+
+  /**
+   * Saves coverage as feature file.
+   * @see FeatureRDD.save
+   *
+   * Supported file formats include bed, narrowPeak and parquet. Coverage is saved
+   * as a feature where coverage is stored in score attribute.
+   *   val chrom    = feature.getContigName
+   *   val start    = feature.getStart
+   *   val end      = feature.getEnd
+   *   val name     = Features.nameOf(feature)
+   *   val coverage = feature.getScore
+   *
+   * @param filePath The location to write the output.
+   */
+  def save(filePath: java.lang.String) = {
+    this.toFeatureRDD.save(filePath)
+  }
+
+  /**
+   * Merges adjacent ReferenceRegions with the same value. This is used for merging coverage.
+   *
+   * @return merged tuples of adjacent ReferenceRegions and coverage
+   */
+  def collapse: CoverageRDD = {
+    val newRDD: RDD[Coverage] = rdd
+      .mapPartitions(iter => {
+        if (iter.hasNext) {
+          val first = iter.next
+          collapse(iter, first, List.empty)
+        } else iter
+      })
+    this.replaceRdd(newRDD)
+  }
+
+  /**
+   * Tail recursion for merging adjacent ReferenceRegions with the same value.
+   * @see merge
+   *
+   * @param iter partition iterator of ReferenceRegion and coverage values
+   * @param lastCoverage: the last coverage from a sorted Iterator that has been considered to merge
+   * @param condensed: Condensed iterator of iter with adjacent regions with the same value merged
+   * @return merged tuples of adjacent ReferenceRegions and coverage
+   */
+  @tailrec private def collapse(iter: Iterator[Coverage],
+                                lastCoverage: Coverage,
+                                condensed: List[Coverage]): Iterator[Coverage] = {
+    if (!iter.hasNext) {
+      // if lastCoverage has not yet been added, add to condensed
+      val nextCondensed =
+        if (condensed.map(r => ReferenceRegion(r)).filter(_.overlaps(ReferenceRegion(lastCoverage))).isEmpty) {
+          lastCoverage :: condensed
+        } else condensed
+      nextCondensed.toIterator
+    } else {
+      val cov: Coverage = iter.next
```

Comment:
Stylistically, I don't love the explicit typing here. Is it needed, or can we let `scalac` infer the types?